pinecone/utils/setup_openapi_client.py

```diff
@@ -1,14 +1,33 @@
-from pinecone.core.client.api_client import ApiClient
 from .user_agent import get_user_agent
+import copy
 
-def setup_openapi_client(api_klass, config, openapi_config, pool_threads):
-    api_client = ApiClient(
+def setup_openapi_client(api_client_klass, api_klass, config, openapi_config, pool_threads, api_version=None, **kwargs):
+    # It is important that we allow the user to pass in a reference to api_client_klass
+    # instead of creating a direct dependency on ApiClient because plugins have their
+    # own ApiClient implementations. Even if those implementations seem like they should
+    # be functionally identical, they are not the same class and have references to
+    # different copies of the ModelNormal class. Therefore cannot be used interchangeably.
+    # without breaking the generated client code anywhere it is relying on isinstance to make
```

Comment:
**super nit:** I think there's an extra period after interchangeably. Thank you for these detailed comments also, helpful for putting the pieces together.